src/toolkits/style_transfer/style_transfer_data_iterator.cpp

```diff
@@ -25,95 +25,82 @@ flex_image get_image(const flexible_type& image_feature) {
   }
 }
 
-std::vector<flexible_type> get_style(const data_iterator::parameters& params) {
-  gl_sarray style_images = params.style[params.style_column_name];
+gl_sarray get_style(const data_iterator::parameters& params) {
+  gl_sarray style_images = params.style;
 
-  if (style_images.dtype() == flex_type_enum::IMAGE) {
-    style_images =
-        style_images.apply(image_util::encode_image, flex_type_enum::IMAGE);
-  }
-
-  gl_sframe result({{params.style_column_name, style_images}});
-
-  std::vector<flexible_type> style_batch;
-  style_batch.reserve(result.size());
+  if (style_images.dtype() == flex_type_enum::IMAGE)
+    style_images = style_images.apply(image_util::encode_image, flex_type_enum::IMAGE);
 
-  for (const auto& i : result.range_iterator()) {
-    const flexible_type& style_image = i[0];
-    style_batch.emplace_back(style_image);
-  }
-
-  return style_batch;
+  return style_images;
 }
 
-gl_sframe get_content(const data_iterator::parameters& params) {
-  gl_sarray content_images = params.style[params.content_column_name];
+gl_sarray get_content(const data_iterator::parameters& params) {
```

Comment:
At this point, it's probably cleanest to unify these into something like `encode_images_if_necessary` or `ensure_encoded` and have the argument be the `gl_sarray`

Reply:
Refactored!